Dockerfile

```diff
@@ -0,0 +1,11 @@
+FROM --platform=$BUILDPLATFORM gradle:jdk17 as builder
+WORKDIR /build
+COPY build.gradle settings.gradle gradle.properties ./
+COPY .git .git
+COPY src src
+RUN gradle --no-daemon build
+
+FROM --platform=$TARGETPLATFORM eclipse-temurin:21-jre-alpine
```

Comment:
The jdk image is about 80 megabytes bigger then the jre, so i think doing that (unnecessarily) would be a bad idea

Reply:
Also could you tell me what those "workflow reasons" you talked about are exactly?

Reply:
For example, The `setup-java` step will use the JDK version instead of JRE regardless of didtribution which is done to prepare for both compiling and runtime.

Reply:
Ok but we dont use setup-java step here, so I don't see how that's related

Reply:
From my understanding, using the jdk to run is entirely unecessary and the jre works just fine. Using the jdk would only make the downloadable image bigger.

Reply:
I also think the JRE is fine for running ViaProxy. The JDK is only needed for building ViaProxy.

Reply:
Yeah, while doing it how im currently doing doubles the bandwidth and storage needed for building, it roughly halves the final image size, which i think is the better outcome. (Also not a lot of people will probably be building the image on their own, as github actions already does that)